libminifi/include/core/MetaInfo.h

```diff
@@ -0,0 +1,180 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef __META_INFO_H__
+#define __META_INFO_H__
+
+#include <algorithm>
+#include <sstream>
+#include <string>
+#include <map>
+#include <mutex>
+#include <atomic>
+#include <functional>
+#include <set>
+#include <stdlib.h>
+#include <math.h>
+#include "utils/StringUtils.h"
+#include "core/FlowFile.h"
+#include "core/state/metrics/DeviceInformation.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace core {
+
+// MetaInfo Class
+class MetaInfo {
+
+ public:
+  // Constructor
+  /*!
+   * Create a new meta info
+   */
+  explicit MetaInfo(const std::string &name, const std::string &value)
+      : name_(name),
+        value_(value) {
+  }
+
+  // Destructor
+  virtual ~MetaInfo() {
+  }
+
+  // Get Name for the meta info
+  std::string getName() {
+    return name_;
+  }
+  // Get value for the meta info, overwritten by specific child meta info
+  virtual std::string getValue(const std::shared_ptr<core::FlowFile> &flow) {
+    return value_;
+  }
+
+ protected:
+  // Name
+  std::string name_;
+  // Value
+  std::string value_;
+
+ private:
+};
+
+#ifdef MINIFI_VERSION
```

Comment:
This appears to be copied from stack overflow. Should probably change this .

Reply:
will do